plugins/tedge_apt_plugin/src/main.rs

```diff
@@ -247,6 +247,27 @@ fn run_cmd(cmd: &str, args: &str) -> Result<ExitStatus, InternalError> {
     Ok(status)
 }
 
+fn get_name_and_version(line: &str) -> (&str, &str) {
+    let vec: Vec<&str> = line.split(' ').collect();
+    // The first element should always exist and has slash, like tedge/tedge-main
+    let name = vec
+        .first()
+        .unwrap_or(&"unknown name")
```

Comment:
It was strange to see string slices like `"unknown name"` being referenced further with `&`s but then I saw that the argument expects `&&str` as the `vec` itself is a `Vec<&str>`.

Reply:
There is a related `clippy::needless_borrow` warning, but two lines below.

```
@@ -254,7 +254,7 @@ fn get_name_and_version(line: &str) -> (&str, &str) {
         .first()
         .unwrap_or(&"unknown name")
         .split_once('/')
-        .unwrap_or((&"unknown name", &"others"))
+        .unwrap_or(("unknown name", "others"))
         .0;
```